include/exadg/incompressible_navier_stokes/postprocessor/line_plot_calculation.cpp

```diff
@@ -64,45 +89,43 @@ LinePlotCalculator<dim, Number>::evaluate(VectorType const & velocity,
     // precision
     unsigned int const precision = data.line_data.precision;
 
-    // loop over all lines
-    for(typename std::vector<std::shared_ptr<Line<dim>>>::const_iterator line =
-          data.line_data.lines.begin();
-        line != data.line_data.lines.end();
-        ++line)
+    LinearAlgebra::distributed::Vector<double> ghosted_pressure(pressure.get_partitioner());
+    ghosted_pressure.copy_locally_owned_data_from(pressure);
+    ghosted_pressure.update_ghost_values();
+
+    std::vector<double> pressure_values =
+      VectorTools::point_values<1>(*evaluation_cache, *dof_handler_pressure, ghosted_pressure);
+
+    LinearAlgebra::distributed::Vector<double> ghosted_velocity(velocity.get_partitioner());
+    ghosted_velocity.copy_locally_owned_data_from(velocity);
+    ghosted_velocity.update_ghost_values();
+
+    std::vector<Tensor<1, dim>> velocity_values =
+      VectorTools::point_values<dim>(*evaluation_cache, *dof_handler_velocity, ghosted_velocity);
```

Comment:
This somehow seems to break the modularity of the code that we had before. Now, we always evaluate velocity and pressure, while - originally - fields have only been evaluated when needed?

Reply:
The background is also the following: For typical flow examples or benchmarks, the velocity is evaluated along 10 lines (transveral to the main flow direction through the domain to obtain velocity profiles), while the pressure is evaluated along a single line (surface of a body in 2d or line along the surface in 3d).